original/java/src/main/java/org/dbpedia/categories/database/DatabaseConnection.java

```diff
@@ -0,0 +1,45 @@
+package org.dbpedia.categories.database;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.sql.*;
+import java.util.Properties;		
+
+/*
+ * This class establishes the Database Connection. 
+*/
+
+public class DatabaseConnection {
+	
+	private static Connection con = null;
+	public DatabaseConnection() {};
+	
+	public static Connection getConnection() {	
+		if(con!=null){
+			return con;
+		}
+		
+		Connection con = null;
+		Properties props= new Properties();
+		FileInputStream file = null;  
+		
+		try {
+			file = new FileInputStream("db.properties");
+			props.load(file);
+
+			// load the Driver Class
+			Class.forName("com.mysql.jdbc.Driver").newInstance();
+
+			// create the connection now	
+			con = DriverManager.getConnection(props.getProperty("DB_URL"),
+					props.getProperty("DB_USERNAME"),
+					props.getProperty("DB_PASSWORD"));
+
+		} catch (IOException | ClassNotFoundException | SQLException | InstantiationException | IllegalAccessException e) {
+			// TODO Auto-generated catch block
```

Comment:
logging is a better option here, for now it is OK but you should mark it as a todo for later. Log4j2 / slf4j is a good option